server/graphql/v1/mutations/orders.js

```diff
@@ -551,6 +551,21 @@ export async function createOrder(order, req) {
         if (get(order, 'paymentMethod.service') === 'stripe') {
           // For Stripe `save` will be manually set to `true`, in `processOrder` if the order succeed
           order.paymentMethod.saved = null;
+
+          // Check if the payment method is already saved
+          const paymentMethod = await models.PaymentMethod.findOne({
+            where: {
+              CollectiveId: fromCollective.id,
+              service: 'stripe',
+              saved: true,
+              data: {
+                fingerprint: order.paymentMethod.data.fingerprint,
+              },
+            },
+          });
+          if (paymentMethod && !order.paymentMethod?.uuid && order.paymentMethod?.save) {
+            await paymentMethod.destroy();
+          }
```

Comment:
What will happen with orders that hold a reference to the old payment method (i.e. other active subscriptions) if we delete it?

Reply:
@Betree : Great point; something I've overlooked. I've just modified the code so that the existing orders (which are still active) are updated with the new payment method. Let me know how it looks; especially the order status I considered I am wondering if I got it precisely.